src/utilities.jl

```diff
@@ -83,20 +83,20 @@ Convert vector data between file formats.
 The output dataset.
 """
 function unsafe_gdalvectortranslate(
-        datasets::Vector{Dataset},
+        datasets::Vector{T},
         options = String[];
         dest = "/vsimem/tmp"
-    )
+    ) where T<:AbstractDataset
     options = GDAL.gdalvectortranslateoptionsnew(options, C_NULL)
     usage_error = Ref{Cint}()
     result = GDAL.gdalvectortranslate(dest, C_NULL, length(datasets),
         [ds.ptr for ds in datasets], options, usage_error)
     GDAL.gdalvectortranslateoptionsfree(options)
-    return Dataset(result)
+    return T(result)
```

Comment:
This should still be `Dataset(result)` rather than `T(result)` (see https://github.com/yeesian/ArchGDAL.jl/pull/167#issuecomment-765704084 for context).